### ISSUE TITLE
KAN cannot fit the simple function y=1/x

### ISSUE THREAD

**Issue by TytTest** [Blealtan/efficient-kan#55]
I simply replaced the objective function in tests/test_Simple_math. py with y=1/x, and the model won't work.
My test code is as follows:

def test_mul():
    kan = KAN([2, 2, 1], base_activation=nn.Identity)
    optimizer = torch.optim.LBFGS(kan.parameters(), lr=0.1)
    with tqdm(range(100)) as pbar:
        for i in pbar:
            loss, reg_loss = None, None

            def closure():
                optimizer.zero_grad()
                x = torch.rand(1024, 2) * 100 + 1e-5
                y = kan(x, update_grid=(i % 20 == 0))

                assert y.shape == (1024, 1)
                nonlocal loss, reg_loss
                u = x[:, 0]
                v = x[:, 1]
                # loss = nn.functional.mse_loss(y.squeeze(-1), (u + v) / (1 + u * v))
                loss = nn.functional.mse_loss(y.squeeze(-1), 1/u)

                reg_loss = kan.regularization_loss(1, 0)
                (loss + 1e-5 * reg_loss).backward()
                return loss + reg_loss

            optimizer.step(closure)
            pbar.set_postfix(mse_loss=loss.item(), reg_loss=reg_loss.item())
    for layer in kan.layers:
        print(layer.spline_weight)

**Comment by TytTest** [comment]
The training result is:
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 100/100 [00:02<00:00, 35.48it/s, mse_loss=3.1, reg_loss=0.00184]

**Comment by notadoc627** [comment]
hi, I encoutered the same question. Did you solve it?
